src/api/emulations/nfs.js

```diff
@@ -168,8 +180,11 @@ class NFS {
   * @returns {Promise<File>} - the same file
   **/
   insert(fileName, file) {
-    return lib.file_insert(this.mData.app.connection, this.mData.ref, fileName, file.ref.ref())
-      .then(() => file);
+    const ffiFile = new File(file);
```

Comment:
`new File` is ensuring that the given file is in the FfiFile format in safe_core and is also a ref type to send it as a reference to the network. 

I'm glad you point this out though because something is missing. I need to check if the file is already a ref type.